packages/iris-grid/src/IrisGridTableModel.js

```diff
@@ -458,35 +460,34 @@ class IrisGridTableModel extends IrisGridModel {
     return '';
   }
 
-  getColumnsWithLayoutHints = memoize((columns, hints) => {
-    if (hints) {
+  getColumnsWithLayoutHints = memoize((columns, hints, userFrozenColumns) => {
+    if (hints || userFrozenColumns.length) {
       const columnMap = new Map();
       columns.forEach(col => columnMap.set(col.name, col));
 
       let frontColumns = [];
       let backColumns = [];
       let frozenColumns = [];
 
-      if (hints.frontColumns) {
+      if (hints?.frontColumns) {
         frontColumns = hints.frontColumns
           .map(name => columnMap.get(name))
           .filter(Boolean);
       }
-      if (hints.backColumns) {
+      if (hints?.backColumns) {
         backColumns = hints.backColumns
           .map(name => columnMap.get(name))
           .filter(Boolean);
       }
-      if (hints.frozenColumns) {
-        frozenColumns = hints.frozenColumns
+      if (userFrozenColumns) {
```

Comment:
You're still not pulling from the layoutHints frozen columns here.

Reply:
We're passing in `this.frozenColumns` here, bad variable naming I think - so it would get LH.frozenColumns.